src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -925,19 +925,24 @@ bool emberAfTimeSynchronizationClusterSetTimeZoneCallback(
         }
         else
         {
-            TimeState dstState = TimeSynchronizationServer::Instance().UpdateDSTOffsetState();
-            TimeSynchronizationServer::Instance().ClearDSTOffset();
-            if (dstState == TimeState::kActive || dstState == TimeState::kChanged)
-            {
-                emitDSTStatusEvent(commandPath.mEndpointId, false);
-            }
             response.DSTOffsetRequired = true;
         }
     }
     else
     {
         response.DSTOffsetRequired = true;
     }
+
+    if (response.DSTOffsetRequired)
+    {
+        TimeState dstState = TimeSynchronizationServer::Instance().UpdateDSTOffsetState();
+        TimeSynchronizationServer::Instance().ClearDSTOffset();
+        if (dstState == TimeState::kActive || dstState == TimeState::kChanged)
```

Comment:
I don't totally understand the enum here - it seems like a mix of states and transitions (active is a state, changed is a transition, but it's not clear from what to what). In what cases would you get back changed vs. active?

Will this also clear the dst offsets if you set the time zone to the same thing it was before?

Reply:
Setting to the same time zone is allowed and considered as new setting - which means dst offset will be cleared.

```
    kInvalid = 0, // No valid offset available
    kActive  = 1, // An offset is currently being used
    kChanged = 2, // An offset expired or changed to a new value
    kStopped = 3, // Permanent item in use
```

kChanged is there (as a state) because we can only tell it changed when some interaction happens, rather than a timer based change. In the latter case we would not have kChanged in there since at the time of an interaction we would already have the correct state.

Reply:
What happens if it was inactive before and it's inactive now?

Reply:
Nothing should happen in that case - kInvalid. My main goal was to trace changes and to decide whether there is a valid time zone or offset at a given time.